float64[] corner_0
float64[] corner_1
float64[] corner_2
float64[] corner_3
---
